core/src/main/java/io/questdb/griffin/engine/functions/str/ReplaceVarcharFunctionFactory.java

```diff
@@ -0,0 +1,212 @@
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2024 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.str;
+
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.CairoException;
+import io.questdb.cairo.sql.Function;
+import io.questdb.cairo.sql.Record;
+import io.questdb.griffin.FunctionFactory;
+import io.questdb.griffin.PlanSink;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.griffin.engine.functions.TernaryFunction;
+import io.questdb.griffin.engine.functions.VarcharFunction;
+import io.questdb.griffin.engine.functions.constants.VarcharConstant;
+import io.questdb.std.IntList;
+import io.questdb.std.ObjList;
+import io.questdb.std.str.Utf8Sequence;
+import io.questdb.std.str.Utf8Sink;
+import io.questdb.std.str.Utf8StringSink;
+import org.jetbrains.annotations.NotNull;
+
+public class ReplaceVarcharFunctionFactory implements FunctionFactory {
+
+    private static final String SIGNATURE = "replace(ØØØ)";
+
+    @Override
+    public String getSignature() {
+        return SIGNATURE;
+    }
+
+    @Override
+    public Function newInstance(
+            int position, ObjList<Function> args, IntList argPositions,
+            CairoConfiguration configuration, SqlExecutionContext sqlExecutionContext
+    ) {
+        final Function replaceWith = args.getQuick(2);
+        if (replaceWith.isConstant() && replaceWith.getVarcharSize(null) < 0) {
+            return VarcharConstant.NULL;
+        }
+        final Function lookFor = args.getQuick(1);
+        if (lookFor.isConstant()) {
+            if (lookFor.getVarcharSize(null) < 0) {
+                return VarcharConstant.NULL;
+            } else if (lookFor.getVarcharSize(null) == 0) {
+                return args.getQuick(0);
+            }
+        }
+
+        final Function value = args.getQuick(0);
+        if (value.isConstant()) {
+            int len = value.getVarcharSize(null);
+            if (len < 1) {
+                return value;
+            }
+        }
+
+        final int maxLength = configuration.getStrFunctionMaxBufferLength();
+        return new Func(value, lookFor, replaceWith, maxLength);
+    }
+
+    private static class Func extends VarcharFunction implements TernaryFunction {
```

Comment:
This function as well as its string analog should have `isReadThreadSafe` return `false`. That's because they use utf-8/string sinks, so we can't use a single instance of these functions for parallel query execution. When this method returns `false`, we clone the function for each worker thread.
```java
    @Override
    public boolean isReadThreadSafe() {
        return false;
    }
```